hooks/terraform_wrapper_module_for_each.sh

```diff
@@ -0,0 +1,409 @@
+        ;;
+      --module-repo-shortname)
+        module_repo_shortname="$value"
+        ;;
+      --module-repo-provider)
+        module_repo_provider="$value"
+        ;;
+      --dry-run)
+        dry_run="true"
+        ;;
+      --verbose)
+        verbose="true"
+        ;;
+      *)
+        echo "ERROR: Unrecognized argument: $key"
+        echo "Hook ID: $HOOK_ID."
+        echo "Generate Terraform module wrapper. Available arguments:"
+        echo "--root-dir=...                - Root dir of the repository (Optional)"
+        echo "--module-dir=...              - Single module directory. Values: \".\" (means just root module), \"modules/iam-user\" (a single module), or empty (means include all submodules found in \"modules/*\"). Default: \"${module_dir}\". (Optional)"
+        echo "--wrapper-dir=...             - Directory where 'wrappers' should be saved. Default: \"${wrapper_dir}\". (Optional)"
+        echo "--module-repo-org=...         - Module repository organization (e.g., 'terraform-aws-modules'). (Optional)"
+        echo "--module-repo-shortname=...   - Short name of the repository (e.g., for 'terraform-aws-s3-bucket' it should be 's3-bucket'). (Optional)"
+        echo "--module-repo-provider=...    - Name of the repository provider (e.g., for 'terraform-aws-s3-bucket' it should be 'aws'). (Optional)"
+        echo "--dry-run                     - Whether to run in dry mode. By default, files will be overwritten."
+        echo "--verbose                     - Show verbose output."
+        echo
+        echo "Example:"
+        echo "--module-dir=modules/object   - Generate wrapper for one specific submodule."
+        echo "--module-dir=.                - Generate wrapper for the root module."
+        echo "--module-repo-org=terraform-google-modules --module-repo-shortname=network --module-repo-provider=google  - Generate wrappers for repository available by name \"terraform-google-modules/network/google\" in the Terraform registry and it includes all modules (root and in \"modules/*\")."
+        exit 1
+        ;;
+    esac
+
+  done
+
+  if [[ -z "$root_dir" ]]; then
+    echo "--root-dir can't be empty. Remove it to use default value."
+    exit 1
+  fi
+
+  if [[ -z "$wrapper_dir" ]]; then
+    echo "--wrapper-dir can't be empty. Remove it to use default value."
+    exit 1
+  fi
+
+  if [[ -z "$module_repo_org" ]]; then
+    echo "--module-repo-org can't be empty. Remove it to use default value."
+    exit 1
+  fi
+
+  if [[ -z "$module_repo_shortname" ]]; then
+    echo "--module-repo-shortname can't be empty. It should be part of full repo name (eg, s3-bucket)."
+    exit 1
+  fi
+
+  if [[ -z "$module_repo_provider" ]]; then
+    echo "--module-repo-provider can't be empty. It should be name of the provider used by the module (eg, aws)."
+    exit 1
+  fi
+
+  if [ ! -d "$root_dir" ]; then
+    echo "Root directory $root_dir does not exist!"
+    exit 1
+  fi
+
+  declare -a all_module_dirs=("./")
```

Comment:
It's kind of superfluous to use `declare` when you assign value to array right away:
```suggestion
  all_module_dirs=("./")
```